crates/lib/src/compiler/quilc.rs

```diff
@@ -302,12 +277,12 @@ impl Error {
 
 /// The response from quilc for a `quil_to_native_quil` request.
 #[derive(Clone, Deserialize, Debug, PartialEq, PartialOrd)]
-struct QuilToNativeQuilResponse {
+pub(crate) struct QuilToNativeQuilResponse {
     /// The compiled program
-    quil: String,
+    pub(crate) quil: String,
     /// Metadata about the compiled program
     #[serde(default)]
-    metadata: Option<NativeQuilMetadata>,
+    pub(crate) metadata: Option<NativeQuilMetadata>,
```

Comment:
It did occur to me that those types (and the QVM variants) might be better suited elsewhere, but hadn't thought about incorporating them into `libquil-sys`. That does make sense to me, though it does add another MR in this train. I can do it quickly and see if it feels like a benefit.

Reply:
Actually, I think depending on libquil-sys at this point is probably more work than it is worth since it is very much still in flux and there are certainly going to be breaking changes there that might bubble up here.